plugins/package-curation-providers/clearly-defined/src/main/kotlin/ClearlyDefinedPackageCurationProvider.kt

```diff
@@ -57,41 +58,33 @@ data class ClearlyDefinedPackageCurationProviderConfig(
     /**
      * The URL of the ClearlyDefined server to use.
      */
+    @OrtPluginOption(defaultValue = "https://api.clearlydefined.io")
     val serverUrl: String,
 
     /**
      * The minimum total score for a curation to be accepted. Must lie within 0 to 100.
      */
+    @OrtPluginOption(defaultValue = "0")
     val minTotalLicenseScore: Int
 )
 
-class ClearlyDefinedPackageCurationProviderFactory :
-    PackageCurationProviderFactory<ClearlyDefinedPackageCurationProviderConfig> {
-    override val type = "ClearlyDefined"
-
-    override fun create(config: ClearlyDefinedPackageCurationProviderConfig) =
-        ClearlyDefinedPackageCurationProvider(config)
-
-    override fun parseConfig(options: Options, secrets: Options) =
-        ClearlyDefinedPackageCurationProviderConfig(
-            serverUrl = options["serverUrl"] ?: Server.PRODUCTION.apiUrl,
-            minTotalLicenseScore = options["minTotalLicenseScore"]?.toInt() ?: 0
-        )
-}
-
 /**
  * A provider for curated package metadata from the [ClearlyDefined](https://clearlydefined.io/) service.
  */
+@OrtPlugin(
+    name = "ClearlyDefined",
+    description = "Provides package curation data from the ClearlyDefined service.",
+    factory = PackageCurationProviderFactory::class
+)
 class ClearlyDefinedPackageCurationProvider(
+    override val descriptor: PluginDescriptor,
     private val config: ClearlyDefinedPackageCurationProviderConfig,
     client: OkHttpClient? = null
 ) : PackageCurationProvider {
-    constructor(serverUrl: String, client: OkHttpClient? = null) : this(
-        ClearlyDefinedPackageCurationProviderConfig(serverUrl, minTotalLicenseScore = 0), client
+    constructor(descriptor: PluginDescriptor, config: ClearlyDefinedPackageCurationProviderConfig) : this(
+        descriptor, config, null
```

Comment:
Can't we still omit `null` from here?

Reply:
No, then the compiler thinks the constructor is trying to call itself.